### ISSUE TITLE
there is something conflictual in between enabling cloudflare rule to log correct ip address in apache for any domain

### ISSUE THREAD

**Issue by lpasquali** [engintron/engintron#972]
hello Fotis and all,

I found that this fix in 1.9.1 :
```
 Fixed Horde webmail forwarding. If you previously tried to open Horde from webmail.domain.tld (either via http:// or https://), cPanel would throw a 404 error. This is now fixed.
```

is not working well with cloudflare rules needed to log correct source IP address in apache.

that's because the only possible IP address information:
```
# Fix Horde webmail forwarding
if ($host ~* "^webmail\.") {
    set $PROXY_FORWARDED_HOST '';
}
```
I do not understand how it should fix the horde problem in #864:

The X-Forwarded-For (XFF) header is a de-facto standard header for identifying the originating IP address of a client connecting to a web server through an HTTP proxy or a load balancer. When traffic is intercepted between clients and servers, server access logs contain the IP address of the proxy or load balancer only. To see the original IP address of the client, the X-Forwarded-For request header is used.
Also is what you use for apache remoteip in your installer:

```
# mod_remoteip (https://httpd.apache.org/docs/current/mod/mod_remoteip.html)
LoadModule remoteip_module modules/mod_remoteip.so
RemoteIPHeader        X-Forwarded-For
RemoteIPInternalProxy $SYSTEM_IPS
EOF
```

if I request a Cloudlfare enabled vhost like www.domain.tld I experience no particular problem but 
when it comes to webmail, users experience sudden disconnects and css instability
Disabling Cloudflare configuration (setting the proxy ip address for vhost) is not possible, it's also optimises requests without checking the dns resolvers. 
I think you should think rolling back the Horde fix related to issue #864 , I also think that issue #965 might be related/merged with this. 
I also forced webmail through https before realizing what I wrote here, following your information on wiki, with no luck

**Comment by fevangelou** [comment]
Do you seriously want to track access to webmail.* subdomains?

**Comment by lpasquali** [comment]
Dear Fotis,
last time I checked (this morning), missing information of webmail logs, for Europe, hits the GDPR laws and might be translated in a 10 million euros fine. If you log stuff in EU you have to keep these accesses as much as GDPR requires and DESTROY them immediately after expiration.

Please do not take as an action moved by sarcasm but I would like to give you this link which is on the subject: 
https://winsolv.com/gdpr-compliance/?gclid=EAIaIQobChMIwt2zivmK3wIVYb7tCh2XQwW8EAAYASAAEgIStvD_BwE

**Comment by hyssop** [comment]
Since #864 is closed, I 'll post this here...

I'm having a loading issue with Horde and SquirrelMail and I'm on 1.9.2 currently. The fix in 1.9.1 works for Horde via HTTP but not SquirrelMail, and does not work for either via HTTPS.

I see in common_http.conf the following fix:

# Fix Horde webmail forwarding
if ($host ~* "^webmail\.") {
    set $PROXY_FORWARDED_HOST '';
}

I placed it in common_https.conf and reloaded nginx but it didn't fix it for HTTPS. (Probably not the correct solution, but I thought I'd try it.)

**Comment by fevangelou** [comment]
@lpasquali If you use CloudFlare then Engintron is your last problem in regards to GDPR...

@hyssop Squirrelmail is end-of-life. If cPanel hasn't removed it in the latest builds they will remove it very soon.
